src/sage/groups/class_function.py

```diff
@@ -123,20 +125,20 @@ def __init__(self, G, values):
         if isinstance(values, GapElement) and gap.IsClassFunction(values):
             self._gap_classfunction = values
         else:
-            self._gap_classfunction = gap.ClassFunction(G, list(values))
+            self._gap_classfunction = gap.ClassFunction(G, gap(values))
```

Comment:
What is the point of calling `gap()` here? This is a call to pexpect GAP, what were getting rid of, right?

Reply:
Well, this is a doctest for `Classfunction_gap`. I have made so that `Classfunction` now calls `ClassFunction_libgap` by default, unless the second argument is a GapPexpect element.

Another possibility would be to import  `Classfunction_gap` and call it explicitely in the doctests.

Of course, it would be better to get rid completely of `Classfunction_gap` but maybe in another ticket ?

Do you have any idea about the failing doctests in class_function.py ?

Reply:
> Do you have any idea about the failing doctests in class_function.py ?

it comes from the fact that GAP operates on characters associated to a character table,
not to a group, and is unable to identify these tables for $\chi_1$ and $\chi_2$ in the 1st failing test. 
Not clear how to fix this easily.

```
sage: chi1 = ClassFunction(G, [3, 1, -1, 0, -1])
sage: isinstance(chi1, ClassFunction_libgap)
True
sage: chi2 = ClassFunction(G, [1, -1, 1, 1, -1])
sage: isinstance(chi2, ClassFunction_libgap)
True
sage: chi1._gap_classfunction
ClassFunction( CharacterTable( Sym( [ 1 .. 4 ] ) ), [ 3, 1, -1, 0, -1 ] )
sage: chi2._gap_classfunction
ClassFunction( CharacterTable( Sym( [ 1 .. 4 ] ) ), [ 1, -1, 1, 1, -1 ] )
sage: chi1._gap_classfunction*chi2._gap_classfunction
---------------------------------------------------------------------------
GAPError                                  Traceback (most recent call last)
Cell In[11], line 1
----> 1 chi1._gap_classfunction*chi2._gap_classfunction

File /mnt/opt/Sage/sage-clang/src/sage/structure/element.pyx:1513, in sage.structure.element.Element.__mul__()
   1511 cdef int cl = classify_elements(left, right)
   1512 if HAVE_SAME_PARENT(cl):
-> 1513     return (<Element>left)._mul_(right)
   1514 if BOTH_ARE_ELEMENT(cl):
   1515     return coercion_model.bin_op(left, right, mul)

File /mnt/opt/Sage/sage-clang/src/sage/libs/gap/element.pyx:1056, in sage.libs.gap.element.GapElement._mul_()
   1054 try:
   1055     sig_GAP_Enter()
-> 1056     sig_on()
   1057     result = GAP_PROD(self.value, (<GapElement>right).value)
   1058     sig_off()

GAPError: Error, no product of class functions of different tables
```